plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/hosted/RecreateMetadataWalkerProcessor.java

```diff
@@ -191,9 +193,21 @@ public void onCollectionExit(final WalkerContext context, final StorageCollectio
       // do this in memory, as package.json is expected to be fairly small (few kb at most)
       final ByteArrayOutputStream bos = new ByteArrayOutputStream(Ints.checkedCast(tarEntry.getSize()));
       StreamSupport.copy(tarArchiveInputStream, bos, StreamSupport.BUFFER_SIZE);
-      final Map<String, Object> packageJson = objectMapper
-          .readValue(bos.toByteArray(), new TypeReference<HashMap<String, Object>>() { });
-      return packageJson;
+
+      try {
+        return objectMapper.readValue(bos.toByteArray(), new TypeReference<HashMap<String, Object>>() { });
```

Comment:
Do we have an input that exhibits the 'bad' behaviour, and can we include that in a unit test?